wp/lib/bap_wp/src/precondition.ml

```diff
@@ -303,14 +304,16 @@ let set_fun_called (post : Constr.t) (env : Env.t) (tid : Tid.t) : Constr.t =
   in
   Constr.substitute_one post fun_name (Bool.mk_true ctx)
 
+(* FIXME: handle other architectures *)
 let increment_stack_ptr (post : Constr.t) (env : Env.t) : Constr.t * Env.t =
   let arch = Env.get_arch env in
   if Env.is_x86 arch then
     begin
-      let module Target = (val target_of_arch arch) in
-      let sp, env = Env.get_var env Target.CPU.sp in
-      let width = Target.CPU.sp |> Var.typ |> typ_size in
-      let addr_size = arch |> Arch.addr_size |> Size.in_bytes in
+      let sp, env = Env.get_sp env |> Env.get_var env in
+      let width = arch |> Theory.Target.bits in
+      (* let addr_size = arch |> Theory.Target.data_addr_size in *)
```

Comment:
Can we delete this section of commented out code?

Reply:
Yes!